src/lib/lessons.ts

```diff
@@ -1,116 +1,149 @@
 import {LessonResource} from 'types'
 import {getGraphQLClient} from '../utils/configured-graphql-client'
 import getAccessTokenFromCookie from '../utils/get-access-token-from-cookie'
+import {loadLessonComments} from './lesson-comments'
+import {sanityClient} from 'utils/sanity-client'
+import groq from 'groq'
 
-export async function loadLesson(slug: string, token?: string) {
-  const query = /* GraphQL */ `
-    query getLesson($slug: String!) {
-      lesson(slug: $slug) {
-        slug
-        title
-        description
-        duration
-        next_up_url
-        free_forever
-        path
-        transcript
-        transcript_url
-        subtitles_url
-        hls_url
-        dash_url
-        http_url
-        media_url
-        lesson_view_url
-        thumb_url
-        icon_url
-        download_url
-        staff_notes_url
-        repo_url
-        code_url
-        created_at
-        updated_at
-        published_at
-        collection {
-          ... on Playlist {
-            title
+const lessonQuery = groq`
+*[_type == 'lesson' && slug == $slug][0]{
+  title,
+  slug,
+  description
+}`
+
+/**
+ * loads LESSON METADATA from Sanity
+ * @param slug
+ */
+async function loadLessonMetadataFromSanity(slug: string) {
+  const params = {
+    slug,
+  }
+
+  const lesson = await sanityClient.fetch(lessonQuery, params)
+
+  return lesson
+}
+
+export async function loadLesson(
+  slug: string,
+  token?: string,
+): Promise<LessonResource> {
+  token = token || getAccessTokenFromCookie()
+  const graphQLClient = getGraphQLClient(token)
+
+  const variables = {
+    slug: slug,
+  }
+
+  const {lesson} = await graphQLClient.request(
+    loadLessonGraphQLQuery,
+    variables,
+  )
+
+  const lessonMetadataFromSanity = await loadLessonMetadataFromSanity(slug)
+
+  const lessonMetadata = {...lesson, ...lessonMetadataFromSanity}
+
+  // comments are user-generated content that must come from the egghead-rails
+  // backend, so load those separately from the rest of lesson metadata.
+  const comments = await loadLessonComments(slug, token)
+
+  return {...lessonMetadata, comments} as LessonResource
```

Comment:
merge in the comments as well

Reply:
suggestion: we might consider making the comments request in a totally separate call from "load lesson" for performance reasons since this adds an additional network call? dunno if it matters but "filling in" comments on the client is probably fine. They might have some SEO advantage.

Reply:
I agree with loading them separately. In this first pass, I was trying to keep the change-surface-area low.